components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/CaptchaApiServiceImpl.java

```diff
@@ -89,21 +90,44 @@ public Response verifyCaptcha(ReCaptchaResponseTokenDTO reCaptchaResponse, Strin
         HttpEntity entity = response.getEntity();
         ReCaptchaVerificationResponseDTO reCaptchaVerificationResponseDTO = new ReCaptchaVerificationResponseDTO();
 
-        try {
+        if (CaptchaConstants.RE_CAPTCHA_TYPE_ENTERPRISE.equals(reCaptchaType)) {
```

Comment:
This can be refactored

```
if (entity == null) {
    RecoveryUtil.handleBadRequest("ReCaptcha verification response is not received.",
            Constants.STATUS_INTERNAL_SERVER_ERROR_MESSAGE_DEFAULT);
}
// For ReCaptcha Enterprise.
try {
    try (InputStream in = entity.getContent()) {
        JsonObject verificationResponse = new JsonParser().parse(IOUtils.toString(in)).getAsJsonObject();
        if (CaptchaConstants.RE_CAPTCHA_TYPE_ENTERPRISE.equals(reCaptchaType)) {
            // For ReCaptcha Enterprise.
            JsonObject tokenProperties = verificationResponse.get(CaptchaConstants.CAPTCHA_TOKEN_PROPERTIES)
                    .getAsJsonObject();
            boolean success = tokenProperties.get(CaptchaConstants.CAPTCHA_VALID).getAsBoolean();
            reCaptchaVerificationResponseDTO.setSuccess(success);
        } else {
            // For ReCaptcha v2 and v3.
            reCaptchaVerificationResponseDTO.setSuccess(verificationResponse.get(
                CaptchaConstants.CAPTCHA_SUCCESS).getAsBoolean());
        }
    }
} catch (IOException e) {
    log.error("Unable to read the verification response.", e);
    RecoveryUtil.handleBadRequest("Unable to read the verification response.",
            Constants.STATUS_INTERNAL_SERVER_ERROR_MESSAGE_DEFAULT);
}
```